terraform crash when using an aliased provider and data block

### Terraform Version

```
0.13.0 beta2 
```

### Terraform Configuration Files

```hcl
provider "azurerm" {
  subscription_id = "zzzzz-zzzzz-zzzzzz-zzzzzz" #redacted
  version         = "~> 2.1.0"
  features {}
}

provider "azurerm" {
  alias           = "platform_kv_subscription" 
  subscription_id = "xxxx-xxxxx-xxxxxx-xxxxxx" #redacted
  version         = "~> 2.1.0"
  features {}
}

data "azurerm_key_vault" "platform_shared" {
  provider                        = "platform_kv_subscription"
  name                             = "myplatformkv"
  resource_group_name = "myresourcegroup"
}
```

### Debug Output
https://gist.github.com/joshschmitter/b881082e6286f1be5adf8fdb7e30c1bf#file-debug_output-log

### Crash Output
https://gist.github.com/joshschmitter/b881082e6286f1be5adf8fdb7e30c1bf#file-crash-log

### Expected Behavior
~~terraform should have downloaded providers and data-block data.~~

*Update*: since the provider reference in the data block was missing the provider type (`azurerm`), terraform should have thrown an error: `Error: no provider exists with the given name`

### Actual Behavior
terraform crashed.

```
panic: must contain only letters, digits, and dashes, and may not use leading or trailing dashes
```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply` 


### Additional Context
The reference to the provider was incorrect, missing the provider type of `azurerm` before the provider alias name.

--- a/configs/resource.go
+++ b/configs/resource.go
@@ -422,6 +422,9 @@ func decodeProviderConfigRef(expr hcl.Expression, argName string) (*ProviderConf
 	name := traversal.RootName()
 	nameDiags := checkProviderNameNormalized(name, traversal[0].SourceRange())
 	diags = append(diags, nameDiags...)
+	if diags.HasErrors() {
+		return nil, diags
+	}
 
 	ret := &ProviderConfigRef{
 		Name:      name,

--- a/configs/testdata/invalid-files/data-invalid-provider-reference.tf
+++ b/configs/testdata/invalid-files/data-invalid-provider-reference.tf
@@ -1,0 +1,3 @@
+data "test_resource" "t" {
+  provider = my_test
+}

--- a/configs/testdata/invalid-files/resource-invalid-provider-reference.tf
+++ b/configs/testdata/invalid-files/resource-invalid-provider-reference.tf
@@ -1,0 +1,3 @@
+resource "test_resource" "t" {
+  provider = my_test
+}

